staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -376,6 +382,38 @@ type CompletedConfig struct {
 	*completedConfig
 }
 
+// AddPostStartHook allows you to add a PostStartHook that will later be added to the server itself in a New call.
+// Name conflicts will cause an error.
+func (c *Config) AddPostStartHook(name string, hook PostStartHookFunc) error {
+	if len(name) == 0 {
+		return fmt.Errorf("missing name")
+	}
+	if hook == nil {
+		return nil
```

Comment:
why not return an error when a hook is empty?

Reply:
+1

At least it has to conflict with another post-start hook. Otherwise the semantics is strange.